### ISSUE TITLE
sentry-tracing: `SentryLayer` ignores `event_filter` if `event_mapper` is set

### ISSUE THREAD

**Issue by jinohkang-theori** [getsentry/sentry-rust#630]
### Environment

Linux x86\_64

### Steps to Reproduce

1. Build `SentryLayer` with both `event_filter` and `event_mapper` set.
   - `event_filter` ignores tracing events below `Level::ERROR`.
   - `event_mapper` always returns `EventMapping::Event(..)`.
3. Emit a tracing event with `Level::WARN` (`warn!()`).

### Expected Result

The `warn` event is discarded by the filter; the Sentry instance does not receive any events.

### Actual Result

The `warn` event is **not** discarded by the filter; the Sentry instance receives the mapped event.

**Comment by Swatinem** [comment]
Yes, this is pretty much intentional. The `event_mapper` takes precedence over the filter, as you can also filter things in it, you just return an `EventMapping::Ignore`.

https://github.com/getsentry/sentry-rust/blob/3e7eec2c27467d528ed3316fe05dc3e5d52f176c/sentry-tracing/src/layer.rs#L136-L138

Though I admit the docs might not fully reflect this.

**Comment by lcian** [comment]
Added this to the docs in https://github.com/getsentry/sentry-rust/pull/839
